pkg/docker/docker.go

```diff
@@ -609,15 +610,30 @@ func (d *stiDocker) RunContainer(opts RunContainerOptions) error {
 
 	containerName := containerNameOrID(container)
 
-	// Container was created, so we defer its removal.
-	defer func() {
+	// Container was created, so we defer its removal, and also remove it if we get a sigterm
+	removalDone := make(chan struct{}, 1)
+	removeContainer := func() {
 		glog.V(4).Infof("Removing container %q ...", containerName)
 		if err := d.RemoveContainer(container.ID); err != nil {
 			glog.Warningf("Failed to remove container %q: %v", containerName, err)
 		} else {
 			glog.V(4).Infof("Removed container %q", containerName)
 		}
+		removalDone <- struct{}{}
+	}
+	go func() {
+		sigchan := make(chan os.Signal, 1)
+		signal.Notify(sigchan, syscall.SIGINT, syscall.SIGTERM, syscall.SIGQUIT)
+		defer signal.Stop(sigchan)
+		defer close(sigchan)
+		select {
+		case <-sigchan:
+			removeContainer()
+			os.Exit(0)
```

Comment:
I mean interrupt.

On Thu, Mar 31, 2016 at 10:25 AM, Ben Parees <notifications@github.com>
wrote:

> In pkg/docker/docker.go
> <https://github.com/openshift/source-to-image/pull/452#discussion_r58062357>
> :
>
> >  		glog.V(4).Infof("Removing container %q ...", containerName)
> >  		if err := d.RemoveContainer(container.ID); err != nil {
> >  			glog.Warningf("Failed to remove container %q: %v", containerName, err)
> >  		} else {
> >  			glog.V(4).Infof("Removed container %q", containerName)
> >  		}
> > +		removalDone <- struct{}{}
> > +	}
> > +	go func() {
> > +		sigchan := make(chan os.Signal, 1)
> > +		signal.Notify(sigchan, syscall.SIGINT, syscall.SIGTERM, syscall.SIGQUIT)
> > +		defer signal.Stop(sigchan)
> > +		defer close(sigchan)
> > +		select {
> > +		case <-sigchan:
> > +			removeContainer()
> > +			os.Exit(0)
>
> (and i agree i found the 0 suspicious)
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/openshift/source-to-image/pull/452/files/8d83b0d6b876181e9d24a1c5b03ac95ca79105fc#r58062357>
>